dashboard/15-final/app/api/auth/[...nextauth]/route.ts

```diff
@@ -0,0 +1,53 @@
+import NextAuth, { NextAuthOptions } from 'next-auth';
+import CredentialsProvider from 'next-auth/providers/credentials';
+import bcrypt from 'bcrypt';
+import { fetchUser } from '../../../lib/data';
+import { User } from '@/app/lib/definitions';
+
+const authOptions: NextAuthOptions = {
+  providers: [
+    CredentialsProvider({
+      name: 'credentials',
+      credentials: {},
+
+      // TODO: Not sure how to type authorize function correctly
+      // @ts-ignore
+      async authorize(credentials: {
+        email: string;
+        password: string;
+      }): Promise<User | null | undefined> {
+        const { email, password } = credentials;
+
+        try {
+          const user = await fetchUser(email);
+
+          if (!user) {
+            return null;
+          }
+
+          const passwordsMatch = await bcrypt.compare(password, user.password);
+
+          if (!passwordsMatch) {
+            console.log('Passwords do not match');
+            return null;
+          }
```

Comment:
I am not a fan of the Credentials provider, to be honest. :grimacing: It has too many footguns. Would prefer if we used an OAuth provider or an Email provider instead. :thinking: